### PR TITLE
Stop referencing {dir}/index.html and only reference {dir}

### PR DESCRIPTION
**Pull request by geeknoid**
This is intended to help improve analytics.

Staging web site: https://geeknoid.github.io/istio.github.io/

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -22,22 +22,22 @@ GEM
     ffi (1.9.18)
     forwardable-extended (2.6.0)
     gemoji (3.0.0)
-    github-pages (139)
+    github-pages (146)
```
Comment (ldemailly) [inline Gemfile.lock]: all those changes just to keep up to date or there is a specific new module/feature ?

I assume "rake test" still works then good!

File: _docs/concepts/policy-and-control/mixer-config.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ This configures an adapter that reports data to the Prometheus system. This adap
 require any custom parameters and so doesn't have a `params` stanza.
 
 Each adapter defines its own particular format of configuration data. The exhaustive set of
-adapters and their specific configuration formats can be found [here]({{home}}/docs/reference/config/mixer/adapters/index.html).
+adapters and their specific configuration formats can be found [here]({{home}}/docs/reference/config/mixer/adapters).
```
Comment (ldemailly) [inline _docs/concepts/policy-and-control/mixer-config.md]: here and everywhere else please add the missing trailing / to save 1 redirect (ideally that would be a lint actually)

File: _includes/doc-side-nav.html [hunk 3]
```diff
@@ -31,7 +31,8 @@
 				{% if d.order == nil %}
 				{% assign order = 9999 %}
 				{% endif %}
-        		], url: "{{d.url}}", title: "{{d.title}}", order: {{order}}, overview: "{{d.overview}}"});
+                {% assign url = d.url | remove: "/index.html" %}
```
Comment (ldemailly) [inline _includes/doc-side-nav.html]: not sure how this works but here also make sure the trailing / is present in the source

**Comment by ldemailly** [comment]
ie you want to avoid:
```
ldemailly-macbookpro:istio ldemailly$ curl -v https://geeknoid.github.io/istio.github.io/docs/reference/config/mixer/adapters
> GET /istio.github.io/docs/reference/config/mixer/adapters HTTP/1.1
> Host: geeknoid.github.io
> User-Agent: curl/7.51.0
> Accept: */*
> 
< HTTP/1.1 301 Moved Permanently
< Server: GitHub.com
< Content-Type: text/html
< Location: http://geeknoid.github.io/istio.github.io/docs/reference/config/mixer/adapters/
```

File: Gemfile.lock (hunk 1 above)
Comment (geeknoid) [inline Gemfile.lock]: This is just getting the latest bits.

**Comment by geeknoid** [comment]
PTAL

**Pull request merged**